libevmasm/Assembly.cpp

```diff
@@ -435,9 +435,8 @@ map<u256, u256> const& Assembly::optimiseInternal(
 	for (size_t subId = 0; subId < m_subs.size(); ++subId)
 	{
 		OptimiserSettings settings = _settings;
-		// Disable creation mode for sub-assemblies.
-		settings.isCreation = false;
-		map<u256, u256> const& subTagReplacements = m_subs[subId]->optimiseInternal(
+		settings.isCreation = m_subs[subId].second;
```

Comment:
If I change this back to hard-coded ``false;``, then I still get ``ffffffffffffffff42`` in the https://github.com/ethereum/solidity/issues/12738#issuecomment-1059158952 setting - it only vanishes with both this and the yul optimizer changes...

But I haven't found a case in which something like this would happen in legacy codegen... aren't created contracts also sub-assemblies for that and should get the wrong value here?

Reply:
In other words: from the looks of this, this should also affect legacy codegen, but I can't reproduce it for that.